framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -587,6 +587,37 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
             this.typeInformationPresenter = null;
         }
 
+        // Alias provided via -AaliasedTypeAnnos command-line option
+        // This can only be used for annotations without attributes,
+        // e.g. this will not be usable to declare an alias for @Regex(2).
+        if (checker.hasOption("aliasedTypeAnnos")) {
+            String aliasesOption = checker.getOption("aliasedTypeAnnos");
+            String[] annos = aliasesOption.split(";");
+            for (String alias : annos) {
+                Pair<Class<? extends Annotation>, @FullyQualifiedName String[]> aliasPair =
+                        parseAliasesFromString(alias);
+                for (@FullyQualifiedName String a : aliasPair.second) {
+                    AnnotationMirror anno = AnnotationBuilder.fromClass(elements, aliasPair.first);
+                    addAliasedTypeAnnotation(a, anno);
+                }
+            }
+        }
+
+        // Alias provided via -AaliasedDeclAnnos command-line option
+        // This can only be used for annotations without attributes,
+        // e.g. this will not be usable to declare an alias for @Regex(2).
+        if (checker.hasOption("aliasedDeclAnnos")) {
+            String aliasesOption = checker.getOption("aliasedDeclAnnos");
+            String[] annos = aliasesOption.split(";");
+            for (String alias : annos) {
+                Pair<Class<? extends Annotation>, @FullyQualifiedName String[]> aliasPair =
+                        parseAliasesFromString(alias);
+                for (String a : aliasPair.second) {
+                    addAliasedDeclAnnotation(a, aliasPair.first);
```

Comment:
Oh, and can't you move the creation of `anno` outside of the loop over `aliasPair.second`, creating the `AnnotationMirror` only once, instead of for each alias again? Not a huge performance concern, but it just seems nicer to do here.